api/query_test.go

```diff
@@ -426,26 +338,18 @@ func TestResponseCompression(t *testing.T) {
 	etreurl := server.url + etre.API_ROOT + "/entities/" + entityType +
 		"?query=" + url.QueryEscape("foo=bar")
 	req, err := http.NewRequest("GET", etreurl, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
 	// Make the request
 	// Note that the http client automatically enables gzip, so we don't have to set the "Accept-Encoding" header.
 	res, err := http.DefaultClient.Do(req)
 	defer res.Body.Close()
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
 	// The http client strips the "Content-Encoding" header so we can't check it directly.
 	// Instead, we have to check the "Uncompressed" flag, which will be *true* if the content came back compressed and was decompressed by the http client.
-	if !res.Uncompressed {
-		t.Errorf("response was not compressed, expected it to be")
-	}
+	assert.True(t, res.Uncompressed, "response was not compressed, expected it to be")
```

Comment:
I think the explanation string would make it clearer to the reader if it said something like:
`response was not decompressed by the http client, we expect it to be`. Currently it just echos the assert statement, which is not very straight forward given the secondary behavior verification.

Reply:
The code comment before this line explains it. I'll update the assert message as well. The only way to tell if the server sent a compressed response is if `res.Uncompressed` is true. so if `res.Uncompressed` is false, then the server did not send a compressed response. We can't check the headers because the client strips them when it does the decompression.